test/response_test.dart

```diff
@@ -25,6 +26,30 @@ void main() {
     });
   });
 
+  test('supports a Uint8List body without copying', () async {
+    var bytes = Uint8List(10);
+    var response = Response.ok(bytes);
+
+    expect(response.contentLength, 10);
+    expect(identical(await response.read().first, bytes), true);
+  });
+
+  test('supports a List<int> body without copying', () async {
+    var bytes = <int>[1, 2, 3, 4];
+    var response = Response.ok(bytes);
+
+    expect(response.contentLength, 4);
+    expect(identical(await response.read().first, bytes), true);
+  });
+
+  test('Copies a dynamic list of int elements', () async {
+    var bytes = <dynamic>[1, 2, 3, 4];
+    var response = Response.ok(bytes);
+
+    expect(response.contentLength, 4);
+    expect(await response.read().first, <int>[1, 2, 3, 4]);
```

Comment:
I'm pretty sure the `<int>` here isn't changing behavior. We're still gonna hit the `equals` matcher which doesn't check generics on collections. If checking the type is an important behavior we'll want to use `isA`

```suggestion
    expect(await response.read().single, isA<List<int>>().having((v) => v, 'content', [1, 2, 3, 4]));
```

Reply:
done